update rust installer with new PATHs

y0rune opened the pull request:
Hello!

I removed the default env loading into a shell config. I use for that a
`sd` and `pathman`.

If something I misunderstood please tell me what I did wrong.

Thank you,
y0rune

Re: #419

coolaj86 commented on rustlang/install.sh:
```diff
@@ -2,9 +2,35 @@
+
 function __install_rust() {
+
+    mkdir -p "$HOME/.cargo/bin"
```
Move these lines 9 and 10 (mkdir and pathman) below the rust installer and that should be good enough.

We shouldn't need to edit .cargo/env.

coolaj86 commented:
You've still got the find and replace going, which should be removed. There's no need to edit the bash file - because fish won't use that.

coolaj86 commented on rustlang/install.sh:
```diff
@@ -2,9 +2,35 @@
+    fi
+
+    echo ""
+    echo 'Removing from shell config '. "$HOME/.cargo/env"''
```
delete all of this stuff. None of this is necessary or useful.

Since `$HOME/.cargo/env` is only for bash, not for fish, it doesn't do any good to edit or remove it.

coolaj86 replied:
delete all of this stuff. None of this is necessary or useful.

Since `$HOME/.cargo/env` is only for bash, not for fish, it doesn't do any good to edit or remove it.

y0rune commented:
Hello!
As far I tested in the docker container with `zsh` and `fish` default shell.

- zsh
```
This path will then be added to your PATH environment variable by
modifying the profile files located at:

  /root/.profile
  /root/.bashrc
  /root/.zshenv
```

```
root@9dc724a2dce7:~# cat .zshenv
. "$HOME/.cargo/env"
```

- fish
```
This path will then be added to your PATH environment variable by
modifying the profile files located at:

  /root/.profile
  /root/.bashrc
```

So I changed the `$ZSHRC` into `$ZSHENV` and removed the fish.

coolaj86 commented:
I don't want any changes to `.cargo/env`.

Adding the `pathman add xxxx` is sufficient and in keeping with how Webi updates PATHs.
